### PR TITLE
Add sv-utils

### PR DESCRIPTION
**Pull request by TedBrookings**
* Add sv-utils python package, which provides support scripts for the GQ filtering pipeline
* Update GQ filtering pipeline WDLs with bugfixes for the transition to WDL version 1.0

### CODE REVIEW

**Comment by TedBrookings** [comment]
Note: the test suite can be executed inside the latest sv-utils docker (us.gcr.io/broad-dsde-methods/tbrookin/sv-utils:8d3d68a). Launch the docker and run `pytest /opt/sv_utils/`
Obviously it can be run locally too, but this takes care of dependency installs. This is a big PR, so I'm happy to do a chat / whatever to explain the logic of it, or discuss problems that need to be fixed.

**Review by mwalker174 — APPROVED** [review]
Thank you @TedBrookings, this represents an enormous amount of work. The sv-utils package can become a model for how we write python in the future, particularly for testing. I have a few comments and suggestions. However, in the interest of time I'm going to approve this so we can move ahead with evaluations, and we can return to this in the near future. One exception would be to delete the "meta" files (see below) since, as far as I can tell, that won't affect functionality. 

- Can you please file a ticket referencing this PR so we can keep track?
- Are `src/sv_utils/meta` and `src/sv_utils/metadata.json` used anywhere?

File: src/sv_utils/setup.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    packages=find_packages("src"),
+    py_modules=[os.path.splitext(os.path.basename(path))[0]
+                for path in glob("src/*.py")],
+    entry_points={
+        "console_scripts": [
+            "sv-utils=sv_utils.command_line:main",
+            "sv_utils=sv_utils.command_line:main"
+        ]
+    },
+    install_requires=[
```
[the remark this answers is not in the document]
Comment (TedBrookings) [reply src/sv_utils/setup.py:26]: Done

File: src/sv_utils/tests/test_genomics_io.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    small_bed_variants = genomics_io.bed_to_pandas(small_bed, literal_eval_columns=test_bed_literal_eval_columns)
+    # note that the order of columns is likely to be different
+    assert not set(small_vcf_variants.columns).symmetric_difference(small_bed_variants.columns)
+    common_test_utils.assert_dataframes_equal(small_bed_variants, small_vcf_variants,
+                                              context=f"vcf={small_vcf}, bed={small_bed}", check_column_order=False)
+    # check that writing the variants to a bed file is byte-identical to the existing bed file. check the
+    # re-ordering mechanism works to create a valid bed file
+    temp_out_bed = os.path.join(temp_out_dir, os.path.basename(small_bed))
+    genomics_io.pandas_to_bed(temp_out_bed, small_vcf_variants)
+    assert not common.command_results(f"zdiff {temp_out_bed} {small_bed}")
```
[the remark this answers is not in the document]
Comment (TedBrookings) [reply src/sv_utils/tests/test_genomics_io.py]: I eliminated calls to zdiff

File: src/sv_utils/src/sv_utils/fix_vcf.py [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+                    else:
+                        warnings.warn(message)
+                        record.info[VcfKeys.bnd_contig_2] = chr2
+
+                record.info[VcfKeys.bnd_end_2] = end2
+
+    if record.pos >= record.stop:
+        if record.pos == record.stop:
+            # empty interval, valid for INS, otherwise an error
+            if record.info[VcfKeys.svtype] not in GAIN_TYPES:
```
[the remark this answers is not in the document]
Comment (TedBrookings) [reply src/sv_utils/src/sv_utils/fix_vcf.py:149]: You're right, I changed it to just INS

File: src/sv_utils/src/sv_utils/fix_vcf.py [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+        else:
+            # sometimes this error occurs for INS or similar SV types
+            warnings.warn(f"{record.id} had END < POS")
+            if flipped_interval_strategy == FlippedIntervalStrategy.end_to_pos:
+                record.stop = record.pos
+            else:
+                # just swap end and pos
+                record.pos, record.stop = record.stop, record.pos
+
+    if record.alts == ('<DUP>',):
```
[the remark this answers is not in the document]
Comment (TedBrookings) [reply src/sv_utils/src/sv_utils/fix_vcf.py:162]: Comment added. It's a legacy problem in some older VCFs

File: src/sv_utils/src/sv_utils/fix_vcf.py [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+    if VcfKeys.bnd_contig_2 in record.info:
+        if record.info[VcfKeys.svtype] in BND_TYPES:
+            if VcfKeys.bnd_end_2 not in record.info:
+                record.info[VcfKeys.bnd_end_2] = record.stop
+            record.stop = record.pos + 1
+        else:
+            # not a break-end, figure out what's going on here...
+            chr2, end2 = str_to_loc(record.info[VcfKeys.source]) \
+                if VcfKeys.source in record.info.keys() else (None, None)
+            if chr2 is None:
```
[the remark this answers is not in the document]
Comment (TedBrookings) [reply src/sv_utils/src/sv_utils/fix_vcf.py:123]: Your way is clearer, I made that change.

**Pull request merged**